component/phlare/scrape/manager.go

```diff
@@ -84,6 +84,8 @@ func (m *Manager) reloader() {
 
 func (m *Manager) reload() {
 	m.mtxScrape.Lock()
+	defer m.mtxScrape.Unlock()
```

Comment:
Why do we keep the lock until all target sets have been synced here? Is it so if a new reload comes in we avoid running new and old sync jobs in parallel? In any case, the scrapePools have their own locks so it might not be needed.

Reply:
yeah you probably right but I went for something safer.